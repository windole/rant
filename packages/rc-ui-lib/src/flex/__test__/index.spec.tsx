import * as React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Flex from '..';

describe('Flex', () => {
  let wrapper;
  afterEach(() => {
    wrapper.unmount();
  });

  it('should render correctly', async () => {
    wrapper = mount(
      <Flex>
        <Flex.Item span={12}>span: 12</Flex.Item>
        <Flex.Item span={12}>span: 12</Flex.Item>
      </Flex>,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should render correctly when using gutter prop', async () => {
    wrapper = mount(
      <Flex gutter={16}>
        <Flex.Item span={8}>span: 8</Flex.Item>
        <Flex.Item span={8}>span: 8</Flex.Item>
        <Flex.Item span={8}>span: 8</Flex.Item>
      </Flex>,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should render correctly when gutter prop is array', async () => {
    wrapper = mount(
      <Flex gutter={[16, -16]}>
        <Flex.Item span={8}>span: 8</Flex.Item>
        <Flex.Item span={8}>span: 8</Flex.Item>
      </Flex>,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should render correctly when direction prop is row-reverse', async () => {
    wrapper = mount(
      <Flex direction="row-reverse">
        <Flex.Item span={8}>span: 8-1</Flex.Item>
        <Flex.Item span={8}>span: 8-2</Flex.Item>
        <Flex.Item span={8}>span: 8-3</Flex.Item>
      </Flex>,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should render correctly when wrap prop is wrap', async () => {
    wrapper = mount(
      <Flex wrap="wrap">
        <Flex.Item span={8}>span: 8-1</Flex.Item>
        <Flex.Item span={8}>span: 8-2</Flex.Item>
        <Flex.Item span={8}>span: 8-3</Flex.Item>
      </Flex>,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should render correctly when using justify prop', async () => {
    wrapper = mount(
      <Flex justify="between">
        <Flex.Item span={8}>span: 8-1</Flex.Item>
        <Flex.Item span={8}>span: 8-2</Flex.Item>
        <Flex.Item span={8}>span: 8-3</Flex.Item>
      </Flex>,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should render correctly when using align prop', async () => {
    wrapper = mount(
      <Flex align="center">
        <Flex.Item span={8}>span: 8-1</Flex.Item>
        <Flex.Item span={8}>span: 8-2</Flex.Item>
        <Flex.Item span={8}>span: 8-3</Flex.Item>
      </Flex>,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should render correctly when using flex prop', async () => {
    wrapper = mount(
      <Flex align="center">
        <Flex.Item flex={2}>span: 8-1</Flex.Item>
        <Flex.Item flex={5}>span: 8-2</Flex.Item>
      </Flex>,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should render correctly when flex prop is decimal', async () => {
    wrapper = mount(
      <Flex align="center">
        <Flex.Item flex="20px">span: 8-1</Flex.Item>
        <Flex.Item flex={5}>span: 8-2</Flex.Item>
      </Flex>,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });
});
