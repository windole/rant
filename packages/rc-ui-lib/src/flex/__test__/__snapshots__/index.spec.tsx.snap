// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex should render correctly 1`] = `
<div
  class="rc-flexbox"
>
  <div
    class="rc-flexitem rc-flexitem--12"
  >
    span: 12
  </div>
  <div
    class="rc-flexitem rc-flexitem--12"
  >
    span: 12
  </div>
</div>
`;

exports[`Flex should render correctly when direction prop is row-reverse 1`] = `
<div
  class="rc-flexbox rc-flexbox--row-reverse"
>
  <div
    class="rc-flexitem rc-flexitem--8"
  >
    span: 8-1
  </div>
  <div
    class="rc-flexitem rc-flexitem--8"
  >
    span: 8-2
  </div>
  <div
    class="rc-flexitem rc-flexitem--8"
  >
    span: 8-3
  </div>
</div>
`;

exports[`Flex should render correctly when flex prop is decimal 1`] = `
<div
  class="rc-flexbox rc-flexbox--align-center"
>
  <div
    class="rc-flexitem"
    style="flex: 0 0 20px;"
  >
    span: 8-1
  </div>
  <div
    class="rc-flexitem"
    style="flex: 5 5 auto;"
  >
    span: 8-2
  </div>
</div>
`;

exports[`Flex should render correctly when gutter prop is array 1`] = `
<div
  class="rc-flexbox"
  style="margin-left: -8px; margin-right: -8px;"
>
  <div
    class="rc-flexitem rc-flexitem--8"
    style="padding-left: 8px; padding-right: 8px;"
  >
    span: 8
  </div>
  <div
    class="rc-flexitem rc-flexitem--8"
    style="padding-left: 8px; padding-right: 8px;"
  >
    span: 8
  </div>
</div>
`;

exports[`Flex should render correctly when using align prop 1`] = `
<div
  class="rc-flexbox rc-flexbox--align-center"
>
  <div
    class="rc-flexitem rc-flexitem--8"
  >
    span: 8-1
  </div>
  <div
    class="rc-flexitem rc-flexitem--8"
  >
    span: 8-2
  </div>
  <div
    class="rc-flexitem rc-flexitem--8"
  >
    span: 8-3
  </div>
</div>
`;

exports[`Flex should render correctly when using flex prop 1`] = `
<div
  class="rc-flexbox rc-flexbox--align-center"
>
  <div
    class="rc-flexitem"
    style="flex: 2 2 auto;"
  >
    span: 8-1
  </div>
  <div
    class="rc-flexitem"
    style="flex: 5 5 auto;"
  >
    span: 8-2
  </div>
</div>
`;

exports[`Flex should render correctly when using gutter prop 1`] = `
<div
  class="rc-flexbox"
  style="margin-left: -8px; margin-right: -8px;"
>
  <div
    class="rc-flexitem rc-flexitem--8"
    style="padding-left: 8px; padding-right: 8px;"
  >
    span: 8
  </div>
  <div
    class="rc-flexitem rc-flexitem--8"
    style="padding-left: 8px; padding-right: 8px;"
  >
    span: 8
  </div>
  <div
    class="rc-flexitem rc-flexitem--8"
    style="padding-left: 8px; padding-right: 8px;"
  >
    span: 8
  </div>
</div>
`;

exports[`Flex should render correctly when using justify prop 1`] = `
<div
  class="rc-flexbox rc-flexbox--justify-between"
>
  <div
    class="rc-flexitem rc-flexitem--8"
  >
    span: 8-1
  </div>
  <div
    class="rc-flexitem rc-flexitem--8"
  >
    span: 8-2
  </div>
  <div
    class="rc-flexitem rc-flexitem--8"
  >
    span: 8-3
  </div>
</div>
`;

exports[`Flex should render correctly when wrap prop is wrap 1`] = `
<div
  class="rc-flexbox rc-flexbox--wrap"
>
  <div
    class="rc-flexitem rc-flexitem--8"
  >
    span: 8-1
  </div>
  <div
    class="rc-flexitem rc-flexitem--8"
  >
    span: 8-2
  </div>
  <div
    class="rc-flexitem rc-flexitem--8"
  >
    span: 8-3
  </div>
</div>
`;
